packages/@okta/vuepress-theme-prose/util/coveo/endpoint.js

```diff
@@ -1,27 +1,43 @@
 import axios from "axios";
+import jwt_decode from "jwt-decode";
+import moment from "moment";
 
 const TOKEN_ENDPOINT =
   "https://wt1ugse0be.execute-api.us-west-2.amazonaws.com/prod/token?site=developer";
 const COVEO_PIPELINE = "oktaproduction9ounvcxa";
 const COVEO_ENDPOINT = "https://platform.cloud.coveo.com/rest/search";
 
-const _getToken = async () => {
-  const {
-    data: { token }
-  } = await axios.get(TOKEN_ENDPOINT);
+const _getToken = () => {
+  const token = localStorage.getItem("coveo_token");
+  if (!token) return;
 
-  localStorage.setItem("coveo_token", token);
-  return token;
+  // Invalidate token if about to expire
+  const decoded = jwt_decode(token);
+  const now = moment();
+  const expWithBuffer = moment.unix(decoded.exp).subtract(5, "minutes");
+  if (now.isAfter(moment(expWithBuffer).subtract(5, "minutes"))) {
```

Comment:
Looks like we are actually doing a buffer of 10 minutes, not sure if it's what you intended but doesn't hurt much either.

Reply:
Whoops, good catch, doesn't hurt but does not look good ^^